bayesian/v2/api_v2.py

```diff
@@ -0,0 +1,248 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Author: Deepak Sharma <deepshar@redhat.com>
+#
+"""Definition of all v2 REST API endpoints of the server module."""
+
+import urllib
+import time
+from requests_futures.sessions import FuturesSession
+from flask import Blueprint, current_app, request, g
+from flask.json import jsonify
+from flask_restful import Api, Resource
+from f8a_worker.utils import (MavenCoordinates, case_sensitivity_transform)
+from fabric8a_auth.auth import login_required
+from bayesian.exceptions import HTTPError
+from bayesian.utils import (get_system_version,
+                            server_create_component_bookkeeping,
+                            server_create_analysis,
+                            check_for_accepted_ecosystem)
+import os
+from fabric8a_auth.errors import AuthError
+from bayesian.v2.utility import VendorAnalyses
+from collections import namedtuple
+
+errors = {
+        'AuthError': {
+                         'status': 401,
+                         'message': 'Authentication failed',
+                         'some_description': 'Authentication failed'
+                     }}
+
+api_v2 = Blueprint('api_v2', __name__, url_prefix='/api/v2')
+rest_api_v2 = Api(api_v2, errors=errors)
+
+
+ANALYSIS_ACCESS_COUNT_KEY = 'access_count'
+TOTAL_COUNT_KEY = 'total_count'
+
+ANALYTICS_API_VERSION = "v2.0"
+HOSTNAME = os.environ.get('HOSTNAME', 'bayesian-api')
+METRICS_SERVICE_URL = "http://{}:{}".format(
+    os.environ.get('METRICS_ACCUMULATOR_HOST', 'metrics-accumulator'),
+    os.environ.get('METRICS_ACCUMULATOR_PORT', '5200')
+)
+
+worker_count = int(os.getenv('FUTURES_SESSION_WORKER_COUNT', '100'))
+_session = FuturesSession(max_workers=worker_count)
+import logging
+_resource_paths = []
+logger = logging.getLogger(__file__)
+
+
+@api_v2.route('/readiness')
+def readiness():
+    """Handle the /readiness REST API call."""
+    return jsonify({}), 200
+
+
+@api_v2.route('/liveness')
+def liveness():
+    """Handle the /liveness REST API call."""
+    # Check database connection
+    current_app.logger.debug("Liveness probe - trying to connect to database "
+                             "and execute a query")
+    return jsonify({}), 200
+
+
+class ApiEndpoints(Resource):
+    """Implementation of / REST API call."""
+
+    @staticmethod
+    def get():
+        """Handle the GET REST API call."""
+        return {'paths': sorted(_resource_paths)}
+
+
+class SystemVersion(Resource):
+    """Implementation of /system/version REST API call."""
+
+    @staticmethod
+    def get():
+        """Handle the GET REST API call."""
+        return get_system_version()
+
+
+class ComponentAnalyses(Resource):
```

Comment:
Code reuse across V1 and V2, may a common module

Reply:
@prashbnair Golden rule of Programing we follow: 
"If its working, don't touch it." :) Actually, as mentioned earlier, idea is to reduce dependency on v1 in any form.

Reply:
I am not sure about sunsetting the API, do we have any timelines. Even otherwise there is some code that is going to be used in all API implementations, that part can definitely be moved out.